### ISSUE TITLE
Feature Request: UI element to multiply an item or a entire category by a factor

### ISSUE THREAD

**Issue by tarexo** [davidhealey/waistline#464]
# Problem
I assume many people tracking calories do not live alone and therefore prepare food for the partners, kids or other family members. Therefore the calories of each ingredient used in the meal is not consumed by the user all by him-/herself, but is divided between multiple people.

Currently the user has two options to avoid tracking more calories than actually consumed:
- Divide the amount of each weighed ingredient while preparing the food
- Go through every used ingredient manually after preparing the food and adjusting the amount of the ingredients as needed

Both methods are not really ideal.

# Solution idea
Therefor I propose a new function in the app, which allows the user to adjust the amounts as needed. There are multiple options to how to implement it.
- Item based: On each item page there is a button, which adjusts the amount of this item only
- Category based: For each category (breakfast, lunch, ...) there is a button in the header of the list, which adjusts each item tracked today under this category

The adjustment itself could be either:
- Flexible: On button press  a dialog opens, prompting the user to insert a value (factor or percentage: 0.5 or 50% respectively),
- Fixed: A drop down menu opens with the (probably) most common factors (25%, 33%, 50%, 66%, 200%). Or just one value. Either way the values could be adjusted to the users needs in the settings

## Related issues
A similar idea can be found in #221 . While I understand the idea, especially regarding pieces of lets say chocolates, the proposed solution would be (in my opinion) more user friendly and offers a improved UX.

**Comment by davidhealey** [comment]
This is what the recipes system is for. Add all your ingredients in there to make your dish, then add the amount of that dish you ate to the diary.

**Comment by tarexo** [comment]
Thanks for the quick response. While this is true for cold dishes like a salad or toast, I do not think recipes are that useful for cooked dishes. 100g of uncooked pasta is not the same as 100g of cooked pasta, as pasta soaks water making them heavier and  therefore reducing the kcal/100g. This is true for all recipes, where water is lost or absorbed (self made bread, pasta, cooked rice, meat and many more come to mind).

**Comment by davidhealey** [comment]
Why not enter the cooked values into the recipe instead of the raw value? Most pasta here in the UK lists the values per 100g and as cooked.

**Comment by EmilJunker** [comment]
> Therefor I propose a new function in the app, which allows the user to adjust the amounts as needed. 

This function already exists, it's called "Number of servings". If you shared the meal with two other people, simply enter a value of `0.33` which effectively divides the weighted amounts by three. You can also do this for an entire recipe.

Turning on the "Prompt for quantity when adding items" option in Diary settings could also help with this.

**Comment by tarexo** [comment]
@EmilJunker yeah, you are correct. This feature solves many of my problems, should've thought early about that! Thanks for your input.

Being able to to a similar operation on whole categories (breakfast, lunch, ...) would be the only feature left I'd suggest. This would speed things up a bit and is a operation I do on a daily basis. If you think this is not worth doing you are free to close this issue (or I close it).

Thanks for the discussion!

**Comment by tarexo** [comment]
@davidhealey 
Interesting! This is sadly not the case for my country (as far as I'm aware). And this still this only applies for one small item. Meat for example could lose up to 30% during cooking, depending on the preparation method.

**Comment by davidhealey** [comment]
> Meat for example could lose up to 30% during cooking

At some point you are going to have to enter the values for the cooked item, so why not enter them in the recipe and forget about the raw values completely since they have no purpose in calorie counting?

**Comment by h-h-h-h** [comment]
There could also be statistical functionality in the app. Regarding cooked vs. raw recipes: You'd just need to be able to input weights before and after cooking, ideally for multiple data points (cooking instances) that are then averaged in the background. This way, the app knows about the weight loss by cooking and can offer to let the user enter cooked weight instead of raw weight.

**Comment by tarexo** [comment]
Might be an interesting idea. I agree that multiple data points should be used to average out any variance in preparation. But you cannot transfer data of one ingredient to another, so it might take some time until you can correctly estimate the loss of a given ingredient. Love the usage of data like this, but i can't imagine many users doing data collection.

For now I will keep weighing my food before cooking it and use the portion sizes to adjust the amount afterwards :+1:
